libgit/repo.go

```diff
@@ -748,3 +751,80 @@ func RenameRepo(
 	}
 	return UpdateRepoMD(ctx, config, tlfHandle, newRepoFS)
 }
+
+// GCOptions describe options foe garbage collection.
+type GCOptions struct {
+	// The most loose refs we will tolerate; if there are more loose
+	// refs, we should pack them.
+	MaxLooseRefs int
+	// TODO: object min age for pruning, and object re-packing thresholds.
+}
+
+func needsGC(storer storage.Storer, options GCOptions) (
+	doPackRefs bool, numLooseRefs int, err error) {
+	numLooseRefs, err = storer.CountLooseRefs()
+	if err != nil {
+		return false, 0, err
+	}
+
+	doPackRefs = numLooseRefs > options.MaxLooseRefs
+	// TODO: check object pruning and object re-packing thresholds.
+	return doPackRefs, numLooseRefs, nil
+}
+
+// GCRepo runs garbage collection on the specified repo, if it exceeds
+// any of the thresholds provided in `options`.
+func GCRepo(
+	ctx context.Context, config libkbfs.Config, tlfHandle *libkbfs.TlfHandle,
+	repoName string, uniqID string, options GCOptions) error {
+	log := config.MakeLogger("")
```

Comment:
Maybe `log := config.MakeLogger("GITGC")`?

Reply:
I personally don't think module labels are useful unless we are going to have multiple instances of something, like we have in the case of the dirty block cache.  In this case, the random context debugging label will uniquely identify requests, and I don't see a need to add a module label.  But if you have a good argument for one, I'll add it!

Reply:
I don't have a good argument, so this sounds good to me. Thanks!